src/metabase/api/user.clj

```diff
@@ -151,11 +153,24 @@
      :offset offset-paging/*offset*}))
 
 
+(defn- with-general-permissions
+  "Adds to `user` a set of boolean flag indiciate whether or not current user has access to a general permissions.
+  These flags will only be added if `advanced-permissions` is enabled."
+  [user]
+  (if-not (premium-features/enable-advanced-permissions?)
+    user
+    (let [permissions-set @api/*current-user-permissions-set*]
+      (assoc user :permissions
+             {:can_access_setting      (perms/set-has-general-permission-of-type? permissions-set :setting)
+              :can_access_subscription (perms/set-has-general-permission-of-type? permissions-set :subscription)
+              :can_access_monitoring   (perms/set-has-general-permission-of-type? permissions-set :monitoring)}))))
```

Comment:
I think this logic can also be in an EE namespace. The function here can just be a wrapper around it, if the EE namespace can be resolved. And the `advanced-permissions` feature flag check should live in EE.